### PR TITLE
571 - Refactor Proxy to accept Files and Append Access Token from Session

### PR DESCRIPTION
**Pull request by joneubank**
## Description of Changes

The Consent-UI needs to proxy web requests through to the Consent-API service with the following requirements:
- replicate initial request including headers/query params/body and properly return response from consent-api to user
- append access token from session (if use has a session with access token)
- correctly send files

This needs to be refactored from the previsouly attempted solutions because:
- NextJS middleware does not allow use of Node APIs; we need either DB access or `crypto` access for decrypting tokens
- Can't use pages router because that does not give access to server-only components (NextAuth)

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/app/consent-api/[...proxy]/route.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		}
+	}
+
+	const session = await auth();
+	const encryptedAccessToken = session?.account?.accessToken;
+
+	if (encryptedAccessToken) {
+		const authorizationheader = `Bearer ${decryptContent(encryptedAccessToken)}`;
+		const participantIdHeader = session.user?.participantId
+			? `,Id ${decryptContent(session.user.participantId)}` // important leading comma needed for appending to string
```
Comment (anncatton) [inline apps/consent-ui/src/app/consent-api/[...proxy]/route.ts:62]: 🙏

File: apps/consent-ui/src/app/consent-api/[...proxy]/route.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import axios, { AxiosError, AxiosHeaders } from 'axios';
+import { NextRequest, NextResponse } from 'next/server';
+import { auth } from 'src/app/auth';
+import { getAppConfig } from 'src/config';
+import { decryptContent } from 'src/services/api/utils';
+
+import urlJoin from 'url-join';
```
Comment (mistryrn) [inline apps/consent-ui/src/app/consent-api/[...proxy]/route.ts:26]: Smol linter change to organize the imports:

```suggestion
import axios, { AxiosError, AxiosHeaders } from 'axios';
import { NextRequest, NextResponse } from 'next/server';
import urlJoin from 'url-join';

import { auth } from 'src/app/auth';
import { getAppConfig } from 'src/config';
import { decryptContent } from 'src/services/api/utils';
```

Comment (joneubank) [reply apps/consent-ui/src/app/consent-api/[...proxy]/route.ts]: 👍 
fixed in another commit.

**Review by anncatton — APPROVED** [review]
thanks @joneubank 🚀

**Pull request merged**